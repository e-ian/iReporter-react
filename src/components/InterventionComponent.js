import React, { Component } from "react";
import "../styles/incidents.scss";

class InterventionComponent extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const { interventions } = this.props;
    const Intervention = interventions.map(incidents => (
      <tr>
        <td>{incidents.comment}</td>
        <td>{incidents.created_by}</td>
        <td>{incidents.created_on}</td>
        <td>{incidents.image}</td>
        <td>{incidents.incident_type}</td>
        <td>{incidents.intervention_id}</td>
        <td>{incidents.location}</td>
        <td>{incidents.status}</td>
      </tr>
    ));
    return (
      <div className="row">
        <div className="col col-lg-1" />
        <div className="col col-lg-10">
          <h2>Interventions</h2>
          <table id="incidentTable">
            <tr className="header">
              <th style={{ width: "10%" }}>Comment</th>
              <th style={{ width: "10%" }}>CreatedBy</th>
              <th style={{ width: "30%" }}>CreatedOn</th>
              <th style={{ width: "10%" }}>Image</th>
              <th style={{ width: "10%" }}>IncidentType</th>
              <th style={{ width: "10%" }}>InterventionId</th>
              <th style={{ width: "15%" }}>Location</th>
              <th style={{ width: "10%" }}>Status</th>
            </tr>
            {Intervention}
          </table>
        </div>
        <div className="col col-lg-1" />
      </div>
    );
  }
}
export default InterventionComponent;
