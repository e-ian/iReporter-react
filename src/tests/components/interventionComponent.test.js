import { shallow } from "enzyme";
import React from "react";
import InterventionComponent from "../../components/InterventionComponent";

describe("get intervention component success tests", () => {
  const props = {
    interventions: [
      {
        comment: "ghjgssbbbababbs",
        created_by: "paulo",
        created_on: "Fri, 31 May 2019 19:10:30 GMT",
        image: "image",
        incident_type: "intervention",
        intervention_id: 17,
        location: "kampala",
        status: "draft"
      },
      {
        comment: "hbbhhbabdfyvweyufey",
        created_by: "paulo",
        created_on: "Fri, 31 May 2019 19:16:35 GMT",
        image: "image",
        incident_type: "intervention",
        intervention_id: 18,
        location: "kampala",
        status: "draft"
      }
    ]
  };
  it("should render without crashing", () => {
    const wrapper = shallow(<InterventionComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
