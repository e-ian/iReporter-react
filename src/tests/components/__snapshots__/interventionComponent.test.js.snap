// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get intervention component success tests should render without crashing 1`] = `ShallowWrapper {}`;
